ckanext/unaids/plugin.py

```diff
@@ -144,6 +149,82 @@ def after_update(self, context, pkg_dict):
                     recipient_org_id=org_to_allow_transfer_to[0]
                 )
 
+    # IResourceController
+    def before_create(self, context, data_dict):
+        print('~'*50)
+        print('before_create')
+        print(data_dict)
+        print('~'*50)
+        return data_dict
+
+    def before_update(self, context, current, data_dict):
+        print('~'*50)
+        print('before_update')
+        print('data_dict')
+        print(data_dict)
+        print('current')
+        print(current)
+
+        def get_upload_authz_token(context, org_name, package_name, resource_id):
+            authorize = toolkit.get_action('authz_authorize')
+            if not authorize:
+                raise RuntimeError("Cannot find authz_authorize; Is ckanext-authz-service installed?")
+            # scope = extstorage_helpers.resource_authz_scope(
+            #     package_name, org_name=org_name, actions='write', resource_id=resource_id)
+            scope = 'obj:ckan/{}/*:write '.format(package_name)
+            print('---> {}'.format(scope))
+            authz_result = authorize(context, {"scopes": [scope]})
+            if not authz_result or not authz_result.get('token', False):
+                raise RuntimeError("Failed to get authorization token for LFS server")
+            if len(authz_result['granted_scopes']) == 0:
+                error = "You are not authorized to upload this resource"
+                log.error(error)
+                raise toolkit.NotAuthorized(error)
```

Comment:
@tomeksabala I've hit a bit of a wall here and could do with some help debugging this
authz keeps returning me no `granted_scopes` meaning it failed

Here's what i'm doing on the front-end https://github.com/fjelltopp/ckanext-unaids/blob/ADX-285-versioning/ckanext/unaids/react/components/BulkFileUploaderComponent/src/App.js#L31-L35

Which successfully returns:
```
{
    "help": "http://dev-adr/api/3/action/help_show?name=authz_authorize",
    "success": true,
    "result": {
        "requested_scopes": [
            "obj:ckan/armenia-inputs-unaids-estimates-2021/*:write"
        ],
        "granted_scopes": [
            "obj:ckan/armenia-inputs-unaids-estimates-2021/*:write"
        ],
        "token": "<TOKEN>",
        "user_id": "admin",
        "expires_at": "2021-04-14T17:09:45.834912+00:00"
    }
}
```

- However in the backend I can't seem to `authorize(context, {"scopes": [scope]})`
- My hunch is that it's related to the `toolkit.c.user` and `toolkit.c.userobj` but they seem valid too